### ISSUE TITLE
[Accessibility]: Commits page - No heading

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#34971]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34423

### Problem description

There is no heading on the page, which means it is difficult for screen readers to navigate around and understand the context of where they are.

### Expected behavior

There should be a H2 heading that describes the page.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
